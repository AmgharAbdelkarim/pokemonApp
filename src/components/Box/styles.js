import styled from 'styled-components';

export const BoxContainer = styled.div`
  box-sizing: border-box;
  height: 56px;
  padding: 20px;
  width: 246px;
  border: 1px solid #979797;
  opacity: 0.24;
  border-radius: 10px;
  cursor: pointer;
  text-align: center;
  margin: 20px;
  > span {
    color: #575e71;
    font-size: 18px;
    font-weight: 500;
    letter-spacing: 0;
    line-height: 16px;
  }
  &:hover {
    opacity: 0.9;
    box-shadow: 0 5px 10px 0 rgba(0, 0, 0, 0.12);
    border: none;
  }
`;
