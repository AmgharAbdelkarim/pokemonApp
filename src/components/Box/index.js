import React from 'react';
import {BoxContainer} from 'components/Box/styles';

const Box = ({name, clickHandler}) => {
  return (
    <BoxContainer onClick={clickHandler}>
      <span>{name}</span>
    </BoxContainer>
  );
};

export default Box;
