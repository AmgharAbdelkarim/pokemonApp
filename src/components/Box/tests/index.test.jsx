import React from "react";

import { cleanup, render } from "@testing-library/react";

import Box from "components/Box";

afterEach(cleanup);

describe("Box", () => {
  it("should Box render correctly", () => {
    const props = {
      clickHandler: jest.fn(),
      name: "fake pokemon",
    };
    const { asFragment, getByText } = render(<Box {...props} />);
    expect(getByText("fake pokemon")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
