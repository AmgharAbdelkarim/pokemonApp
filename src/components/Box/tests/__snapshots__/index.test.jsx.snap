// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box should Box render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ hkRHVq"
  >
    <span>
      fake pokemon
    </span>
  </div>
</DocumentFragment>
`;
